addons/mail/models/ir_model_fields.py

```diff
@@ -30,3 +31,26 @@ def _instanciate_attrs(self, field_data):
         if attrs and field_data.get('tracking'):
             attrs['tracking'] = field_data['tracking']
         return attrs
+
+    def unlink(self):
+        """ When unlinking fields populate tracking value table with relevant
+        information. That way if a field is removed (custom tracked, migration
+        or any other reason) we keep the tracking and its relevant information.
+        Do it only when unlinking fields so that we don't duplicate field
+        information for most tracking. """
+        tracked = self.filtered('tracking')
+        if tracked:
+            tracking_values = self.env['mail.tracking.value'].search(
+                [('field_id', 'in', tracked.ids)]
+            )
+            field_to_trackings = groupby(tracking_values, lambda track: track.field_id)
+            for field, trackings in field_to_trackings:
+                self.env['mail.tracking.value'].concat(*trackings).write({
+                    'field_info': {
+                        'desc': field.field_description,
+                        'name': field.name,
+                        'sequence': self.env[field.model_id.model]._mail_track_get_field_sequence(field.name),
+                        'type': field.ttype,
+                    }
```

Comment:
So this field_info takes a snapshot of the few needed information and store them in a relaxed/application-only.
I think it is a good usage of the new Json field (first time I see it in action)
Regarding the documentation in fields.py, there are some important limitations though, this field should be considered as an application-only fields (like non-stored computed fields), it is not searchable, not groupable, not anything-sql-able. I don't think we'll need to do those operations but IMO it is important that we're well aware of those limitations

Reply:
Indeed, there shouldn't be any search based on field information. Moreover tracking model is limited to admins, so there is no real UI to manipulate them (and thus no sort, groupby, ... targeting end users).